underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -402,6 +410,7 @@ private String getChildName(String child, String parent) {
   private StorageObject getObjectDetails(String key) {
     try {
       if (isFolder(key)) {
+        key = stripSlashIfPresent(key);
         String keyAsFolder = convertToFolderName(stripPrefixIfPresent(key));
```

Comment:
Should the `stripSlashIfPresent` logic go into the `convertToFolderName` method?